lib/search-suggestions/index.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import SmallSearchIcon from '../icons/search-small';
+import TagIcon from '../icons/tag';
+
+export class SearchSuggestions extends Component {
+  static displayName = 'SearchSuggestions';
+
+  static propTypes = {
+    onSearch: PropTypes.func.isRequired,
+    query: PropTypes.string.isRequired,
+    filteredTags: PropTypes.array.isRequired,
+    storeKeyHandler: PropTypes.func.isRequired,
+  };
+
+  state = {
+    selectedItem: 0,
+  };
+
+  componentDidMount() {
+    this.props.storeKeyHandler({
+      next: this.nextItem,
+      prev: this.prevItem,
+      select: this.selectItem,
+    });
+  }
+
+  componentDidUpdate() {
+    // scroll selected item into view
+    var selectedItem = this.refs.selectedItem;
+    var domNode = null;
+    // todo scroll only when needed (i.e. item is not visible)
+    if (selectedItem) {
+      domNode = ReactDOM.findDOMNode(selectedItem);
+      domNode.scrollIntoView(false);
+    }
+
+
+    this.props.storeKeyHandler({
+      next: this.nextItem,
+      prev: this.prevItem,
+      select: this.selectItem,
+    });
+  }
+
+  nextItem = () => {
+    const { filteredTags } = this.props;
+    const { selectedItem } = this.state;
+    const newItem = Math.min(filteredTags.length, selectedItem + 1);
+    this.setState({ selectedItem: newItem });
+  };
+
+  prevItem = () => {
+    const { selectedItem } = this.state;
+    const newItem = Math.max(0, selectedItem - 1);
+    this.setState({ selectedItem: newItem });
+  };
```

Comment:
since these are small and next to each other it's not a big problem but they do share a fair amount of logic and I find that it's a wee bit unclear due to the syntax noise to tell what we're intending on doing. extracting out the common operation (adjust the index) could remove some of that noise and bring more clarity and enforce common behavior

```js
moveSelection = ( direction = 'up' ) => {
	const { filteredTags } = this.props;
	const { selectedItem } = this.state;
	const step =
		'up' === direction ? -1 :
		'down' === direction ? 1 :
		0; // should only be up or down
	const newItem = Math.max( 0, Math.min(filteredTags.length, selectedItem + step) );
	this.setState({ selectedItem: newItem });
}

nextItem = () => this.moveSelection('down');

prevItem = () => this.moveSelection('up');
```